src/main/java/com/amazon/opendistroforelasticsearch/sql/utils/SQLFunctions.java

```diff
@@ -973,34 +974,27 @@ public String getCastScriptStatement(String name, String castType, List<KVValue>
      * approach will return type of result column as DOUBLE, although there is enough information to understand that
      * it might be safely treated as INTEGER.
      */
-    public static Schema.Type getScriptFunctionReturnType(Field field) {
+    public static Schema.Type getScriptFunctionReturnType(
+            int fieldIndex, MethodField field, ColumnTypeProvider scriptColumnType) {
+        Schema.Type returnType;
         String functionName = ((ScriptMethodField) field).getFunctionName().toLowerCase();
+        if (!numberOperators.contains(functionName) && !mathConstants.contains(functionName)
+                && !trigFunctions.contains(functionName) && !stringOperators.contains(functionName)
+                && !stringFunctions.contains(functionName) && !binaryOperators.contains(functionName)
+                && !dateFunctions.contains(functionName) && !conditionalFunctions.contains(functionName)
+                && !utilityFunctions.contains(functionName)) {
+            throw new UnsupportedOperationException(
```

Comment:
Is it possible to handle the unsupported case in SemanticAnalyzer?

Reply:
good catch - looks like SemanticAnalyzer checks for unsupported functions